lib/istoredp.js

```diff
@@ -60,33 +60,45 @@ const db2Call = (callback,xlib,xdatabase,xuser,xpassword,xipc,xctl,xml_input_scr
         [xml_input_script, db.SQL_PARAM_INPUT, 0],
         [xmlOut, db.SQL_PARAM_OUTPUT, 0],
       ];
+
+    const cleanup = () => {
+      stmt.close();
+      conn.disconn();
+      conn.close();
+    };
         
     if(sync == true) {  // Sync Mode
       stmt.prepareSync(sql);

```

Comment:
We should pass a callback to [prepareSync](https://github.com/IBM/nodejs-idb-connector/tree/master/docs#dbstmtpreparesyncsql--callback) and check for an error. Then run `cleanup()` if an error occured.

Reply:
I think wrapping with try/catch would be better.

Reply:
Yes that would make more sense!